File: src/app/modules/m6-StatisticsReports/data-access/models/school-financial-summary-report.dto.ts
import { BaseAuditFields, ApprovalStatus, AuditStatus, FinancialReportType } from './base.types';

export interface SchoolFinancialSummaryReport extends BaseAuditFields {
  schoolId: number;
  fiscalYear: string;
  reportDate: string;
  reportType: FinancialReportType;
  totalBookValue: number;
  totalDepreciation: number;
  totalAssetsCount: number;
  totalAcquisitionCost: number;
  totalRevaluationGains: number;
  totalImpairmentLosses: number;
  totalRevenue: number;
  totalExpenses: number;
  netIncome: number;
  auditStatus: AuditStatus | null;
  auditFirmName: string | null;
  auditDate: string | null;
  approvalStatus: ApprovalStatus;
  approvedByUserId: number | null;
  approvalDate: string | null;
  filePath: string | null;
  notes: string | null;
}

export type CreateSchoolFinancialSummaryReport = Omit<SchoolFinancialSummaryReport, 'id' | 'createdAt' | 'modifiedAt'>;

export type UpdateSchoolFinancialSummaryReport = Pick<SchoolFinancialSummaryReport, 'id'> & Partial<Omit<SchoolFinancialSummaryReport, 'id' | 'createdAt' | 'modifiedAt'>>;

File: 01_Backend/EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/SchoolFinancialSummaryReports/SchoolFinancialSummaryReportDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.SchoolFinancialSummaryReports;

public class SchoolFinancialSummaryReportDto
{
    public long Id { get; set; }
    public long SchoolId { get; set; }
    public string FiscalYear { get; set; } = string.Empty;
    public DateTime ReportDate { get; set; }
    public int ReportType { get; set; }
    public decimal TotalBookValue { get; set; }
    public decimal TotalDepreciation { get; set; }
    public int TotalAssetsCount { get; set; }
    public decimal TotalAcquisitionCost { get; set; }
    public decimal TotalRevaluationGains { get; set; }
    public decimal TotalImpairmentLosses { get; set; }
    public decimal TotalRevenue { get; set; }
    public decimal TotalExpenses { get; set; }
    public decimal NetIncome { get; set; }
    public string? AuditStatus { get; set; }
    public string? AuditFirmName { get; set; }
    public DateTime? AuditDate { get; set; }
    public int ApprovalStatus { get; set; } = 1;
    public long? ApprovedByUserId { get; set; }
    public DateTime? ApprovalDate { get; set; }
    public string? FilePath { get; set; }
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}
